src/realm/object-store/sync/sync_manager.hpp

```diff
@@ -222,23 +222,23 @@ class SyncManager : public std::enable_shared_from_this<SyncManager> {
     // Immediately closes any open sync sessions for this sync manager
     void close_all_sessions() REQUIRES(!m_mutex, !m_session_mutex);
 
+    struct RestartSessionsTag {};
+
+    using RestartSessions = util::TaggedBool<RestartSessionsTag>;
     // Used by App to update the sync route any time the location info has been refreshed.
-    // m_sync_route starts out as unset when the SyncManager is created or configured.
-    // It will be updated to a valid value upon the first App AppServices HTTP request or
-    // the access token will be refreshed (forcing a location update) when a SyncSession
-    // is activated and it is still unset. This value is not allowed to be reset to
-    // nullopt once it has a valid value.
-    void set_sync_route(std::string sync_route) REQUIRES(!m_mutex)
-    {
-        REALM_ASSERT(!sync_route.empty());
-        util::CheckedLockGuard lock(m_mutex);
-        m_sync_route = std::move(sync_route);
-    }
+    // m_sync_route starts out as a generated value based on the configured base_url when
+    // the SyncManager is created by App. If this is incorrect, the websocket connection
+    // will fail, resulting in an update to the access token (and the location, if it hasn't
+    // been updated yet). If the sync route is being manually set without being validated,
+    // then set validated to false.
+    void set_sync_route(std::string sync_route, bool verified = true,
+                        RestartSessions restart_sessions = RestartSessions{true})
```

Comment:
why not just use a bool?

Reply:
Used a tagged bool for `restart_sessions` to help reduce confusion regarding which bool was which...
I actually added this when I was trying to track down the deadlock in case it was from restarting the sessions. Instead it turned out to be in the `TestingStateMachine` used by the tests.